std/experimental/checkedint.d

```diff
@@ -0,0 +1,2678 @@
+/**
+
+This module defines facilities for efficient checking of integral operations
+against overflow, casting with loss of precision, unexpected change of sign,
+etc. The checking (and possibly correction) can be done at operation level, for
+example $(LREF opChecked)$(D !"+"(x, y, overflow)) adds two integrals `x` and
+`y` and sets `overflow` to `true` if an overflow occurred. The flag `overflow`
+(a `bool` passed by reference) is not touched if the operation succeeded, so the
+same flag can be reused for a sequence of operations and tested at the end.
+
+Issuing individual checked operations is flexible and efficient but often
+tedious. The $(LREF Checked) facility offers encapsulated integral wrappers that
+do all checking internally and have configurable behavior upon erroneous
+results. For example, `Checked!int` is a type that behaves like `int` but aborts
+execution immediately whenever involved in an operation that produces the
+arithmetically wrong result. The accompanying convenience function $(LREF
+checked) uses type deduction to convert a value `x` of integral type `T` to
+`Checked!T` by means of `checked(x)`. For example, $(D checked(1_000_000) *
+10_000) aborts execution because the operation overflows. Also, $(D checked(-1) >
+uint(0)) aborts execution (even though the built-in comparison $(D int(-1) >
+uint(0)) is surprisingly true due to language's conversion rules modeled  after
+C). Thus, `Checked!int` is a virtually drop-in replacement for `int` useable in
+debug builds, to be replaced by `int` in release mode if efficiency demands it.
+
+`Checked`  has customizable behavior with the help of a second type parameter,
+`Hook`. Depending on what methods `Hook` defines, core operations on the
+underlying integral may be verified for overflow or completely redefined. If
+`Hook` defines no method at all and carries no state, there is no change in
+behavior, i.e. $(D Checked!(int, void)) is a wrapper around `int` that adds no
+customization at all.
+
+This module provides a few predefined hooks (below) that add useful behavior to
+`Checked`:
+
+$(UL
+
+$(LI $(LREF Abort) fails every incorrect operation with a message to $(REF
+stderr, std, stdio) followed by a call to `std.core.abort`. It is the default
```

Comment:
`std.core.abort`? I can't find any such function (or package).

The implementation of `Abort` below is still using `assert(0)`.